File: src/main/java/org/example/kafka/TestProducer.java
package org.example.kafka;

import io.micronaut.configuration.kafka.annotation.KafkaClient;
import io.micronaut.configuration.kafka.annotation.KafkaKey;
import io.micronaut.configuration.kafka.annotation.Topic;

@KafkaClient("test-producer")
public interface TestProducer {

    @Topic("test_topic")
    public void send(@KafkaKey String key, String value);

}

File: src/main/kotlin/org/example/testing/TestProducer.kt
package org.example.testing

import io.micronaut.configuration.kafka.annotation.KafkaClient
import io.micronaut.configuration.kafka.annotation.KafkaKey
import io.micronaut.configuration.kafka.annotation.Topic

@KafkaClient("test-producer")
interface TestProducer {

    @Topic("test_topic")
    fun send(@KafkaKey key: String, value: String)

}